scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,349 @@
+
+    @staticmethod
+    def _get_bagit_files_from_s3(
+        bucket, path, version, working_folder, tagmanifest_files
+    ):
+        file_locations = [
+            f"{path}/{version}/{file['name']}" for file in tagmanifest_files
+        ]
+
+        for location in file_locations:
+            filename = location.split("/")[-1]
+            save_path = f"{working_folder}/{filename}"
+            storage_s3_client.download_file(bucket, location, save_path)
+
+    @staticmethod
+    def _append_archivematica_uuid(working_folder, archivematica_uuid):
+        bag_info = load_space_separated_file(
+            file_location=f"{working_folder}/bag-info.txt"
+        )
+
+        if "Internal-Sender-Identifier:" in bag_info:
+            assert (
+                bag_info["Internal-Sender-Identifier:"] == archivematica_uuid
+            ), archivematica_uuid
+            return False
+        else:
+            with open(f"{working_folder}/bag-info.txt", "a") as fp:
+                fp.write(f"Internal-Sender-Identifier: {archivematica_uuid}\n")
+            return True
+
+    def _update_tagmanifest(self, working_folder):
+        existing_checksums = self._load_existing_checksums(working_folder)
+        new_checksums = self._generate_updated_checksums(working_folder)
+
+        merged_checksums = dict(existing_checksums)
+        for k, v in new_checksums.items():
+            merged_checksums[k] = v
+
+        assert "fetch.txt" in merged_checksums.keys()
+
+        old_bag_info_checksum = existing_checksums.get("bag-info.txt")
+        new_bag_info_checksum = merged_checksums.get("bag-info.txt")
+
+        assert old_bag_info_checksum != new_bag_info_checksum
+
+        with open(f"{working_folder}/{self.tagmanifest_name}", "w") as fp:
+            for checksum, filename in merged_checksums.items():
+                fp.write(f"{filename} {checksum}\n")
+
+    def _upload_bag_to_s3(self, archive_location, working_id, remove_bag=True):
+        s3_upload_key = f"{self.s3_upload_prefix}/{working_id}.tar.gz"
+        workflow_s3_client.upload_file(
+            Filename=archive_location, Bucket=s3_upload_bucket, Key=s3_upload_key
+        )
+        if remove_bag:
+            os.remove(archive_location)
+
+        return s3_upload_key
+
+    def migrate(self, version, space, external_identifier):
+        storage_manifest = storage_client.get_bag(
+            space=space, external_identifier=external_identifier, version=version
+        )
+
+        id = storage_manifest["id"]
+        bucket = storage_manifest["location"]["bucket"]
+        path = storage_manifest["location"]["path"]
+        files = storage_manifest["manifest"]["files"]
+        provider = storage_manifest["location"]["provider"]["id"]
+        tagmanifest_files = storage_manifest["tagManifest"]["files"]
+
+        assert provider == "amazon-s3"
+
+        working_id = id.replace("/", "_")
+        working_folder = f"{self.target_folder}/{working_id}"
```

Comment:
minor: prefer `os.path.join(…)` to combining paths directly. Not an issue here, but bit more portable in general.